### DISCUSSION TITLE
Help wanted for weird issue with lowdb v2

### DISCUSSION THREAD

**Discussion by amsam0** [vercel/next.js#26846] [Help] [answered]
Hi everyone,

I recently updated the `lowdb` npm package to v2 (https://www.npmjs.com/package/lowdb) and I am getting a weird error. I am not sure if I should ask this in next.js or lowdb but I'll start with next.js

Here's the error:
```
error - ./node_modules/lowdb/lib/adapters/TextFile.js:1:0
Module not found: Can't resolve 'fs'
null
```

There is also another error message that does not consistently show up:
```
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: .\node_modules\lowdb\lib\index.js
require() of ES modules is not supported.
require() of .\node_modules\lowdb\lib\index.js from .\.next\server\pages\index.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename .\node_modules\lowdb\lib\index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from .\node_modules\lowdb\package.json.

    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1102:13)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:14)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:92:18)
    at Object.lowdb (.\.next\server\pages\index.js:770:18)    
    at __webpack_require__ (.\.next\server\webpack-runtime.js:25:42)
    at Object../lib/db.ts (.\.next\server\pages\index.js:262:63)
    at __webpack_require__ (.\.next\server\webpack-runtime.js:25:42)
    at Object../lib/util.ts (.\.next\server\pages\index.js:416:61) {
  code: 'ERR_REQUIRE_ESM'
}
```

lowdb should only be loaded in api handlers and `getServerSideProps()` but I may be missing something here.

Does anyone know how to fix this error?

Thanks in advance!

**Comment by wwwdepot** [comment]
Lowdb is a pure ESM package (from its readme). You need to use https://www.npmjs.com/package/next-transpile-modules

**Reply by amsam0** [reply]
Thanks! I think it solved the second error but this error this occurs:
```
error - ./node_modules/lowdb/lib/adapters/TextFile.js:1:0
Module not found: Can't resolve 'fs'
null
Could not find files for / in .next/build-manifest.json
Could not find files for / in .next/build-manifest.json
~``

**Reply by amsam0** [reply]
I was able to get a `console.log(typeof window);` in and it seems window is undefined which means we are in the server... I'll look into fixing it on the lowdb side of things.

**Comment by amsam0** [comment, marked as answer]
Along with https://github.com/vercel/next.js/discussions/26846#discussioncomment-957174: 
> Lowdb is a pure ESM package (from its readme). You need to use https://www.npmjs.com/package/next-transpile-modules

I managed to solve this by using dynamic imports in getServerSideProps() and I also moved a sneaky import that I didn't find before into the server side code.
